src/sql/workbench/contrib/notebook/test/browser/notebookContexts.test.ts

```diff
@@ -0,0 +1,129 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as TypeMoq from 'typemoq';
+import * as assert from 'assert';
+import { nb } from 'azdata';
+import { NotebookContexts } from 'sql/workbench/contrib/notebook/browser/models/notebookContexts';
+import { IConnectionManagementService } from 'sql/platform/connection/common/connectionManagement';
+import { TestConnectionManagementService } from 'sql/platform/connection/test/common/testConnectionManagementService';
+import { ConnectionProfile } from 'sql/platform/connection/common/connectionProfile';
+import { TestCapabilitiesService } from 'sql/platform/capabilities/test/common/testCapabilitiesService';
+import { mssqlProviderName } from 'sql/platform/connection/common/constants';
+import { IDefaultConnection } from 'sql/workbench/contrib/notebook/browser/models/modelInterfaces';
+
+suite('Notebook Contexts', function (): void {
+	const defaultContext = NotebookContexts.DefaultContext;
+	const localContext = NotebookContexts.LocalContext;
+
+	function createTestConnProfile(): ConnectionProfile {
+		return new ConnectionProfile(new TestCapabilitiesService(), {
+			connectionName: 'Test',
+			savePassword: false,
+			groupFullName: 'testGroup',
+			serverName: 'testServerName',
+			databaseName: 'testDatabaseName',
+			authenticationType: 'integrated',
+			password: 'test',
+			userName: 'testUsername',
+			groupId: undefined,
+			providerName: mssqlProviderName,
+			options: {},
+			saveProfile: true,
+			id: 'testId'
+		});
+	}
+
+	test('Get Contexts For Kernel', async function (): Promise<void> {
+		const connService: TypeMoq.Mock<IConnectionManagementService>
+			= TypeMoq.Mock.ofType<IConnectionManagementService>(TestConnectionManagementService, TypeMoq.MockBehavior.Strict);
+
+		// No kernel or profile info provided
+		let conns = NotebookContexts.getContextsForKernel(connService.object, [mssqlProviderName]);
+		assert.deepEqual(conns, defaultContext);
+
```

Comment:
You should be using deepStrictEqual here - deepEqual is deprecated and it only does == comparison which isn't what we want to do here anyways.